test/books.test.js

```diff
@@ -1,71 +1,150 @@
-  it('Books Post with Alice', async () => {
-
-    const { status, data } = await POST('/odata/v4/catalog/Books', newBook);
+  it('Books Post with Fahd', async () => {
+    const { status, data } = await POST(url('Books'), newBook);
     expect(status).to.equal(201);
     expect(data.author).to.equal(newBook.author);
-    id = data.ID;  
+    bookTestId = data.ID;
   })
 
-  it('Test GET After Post',async()=>{
-    const { data } = await GET(`/odata/v4/catalog/Books(${id})`);
+  it('Test GET After Post', async () => {
+    const { data } = await GET(url('Books', bookTestId));
     expect(data.title).to.eql(newBook.title);
     expect(data.author).to.eql(newBook.author);
-    expect(data.ID).to.eq(id);
+    expect(data.ID).to.eq(bookTestId);
   })
 
-  it('Test Count item in GET BOOKS',async()=>{
-    const { data } = await GET('/odata/v4/catalog/Books');
+  it('Test Count item in GET BOOKS', async () => {
+    const { data } = await GET(url('Books'));
     expect(data.value.length).to.eql(4);
   })
 
   // check unauthenticated user get a 401 status with POST
   it('Books POST with unauthenticated user -> 401', async () => {
 
-    try {
-      await POST('/odata/v4/catalog/Books', newBook , {auth : null});
-    } catch (err) {
-      expect(err.status).to.equal(401);
-    }
+    await executeRequestExpectingErrorStatus(
+      () => POST(url('Books'), newBook, { auth: null }),
+      401,
+      'on POST Book'
+    )
+
   })
 
 
   // check unauthenticated user get a 401 status with GET
   it('Books Get With  unauthenticated user -> 401', async () => {
-    try {
-      await GET('/odata/v4/catalog/Books', { auth : null });
-    } catch (err) {
-      expect(err.status).to.eql(401);
-    }
+
+    await executeRequestExpectingErrorStatus(
+      () => GET(url('Books'), { auth: null }),
+      401, 'on GET Book'
+    );
+
   })
 
   // check PUT Books is rejected with 405
   it('Book PUT Test ->405', async () => {
-    try {
-      await PUT('/odata/v4/catalog/Books(6b4e801c-c081-440d-8353-457767a327a4)', { title: "ana", author: '8' });
-      expect.fail('It should fail here on PUT');
-    } catch (error) {
-      expect(error?.status).to.eql(405, error.message);
-    }
+
+    await executeRequestExpectingErrorStatus(
+      () => PUT(url('Books', bookTestId), { title: "ana", author: '8' }),
+      405,
+      'on PUT Book'
+    )
   })
 
   // check Delete Books is rejected with 405
   it('Book DELETE Test ->405', async () => {
-    try {
-      await DELETE('/odata/v4/catalog/Books(7812e4c4-9db5-4176-bb64-1b216bb2f742)');
-      expect.fail('It should fail here on DELETE');
-    } catch (error) {
-      expect(error?.status).to.eql(405, error.message);
-    }
+
+    await executeRequestExpectingErrorStatus(
+      () => DELETE(url('Books', bookTestId)),
+      405,
+      'on DELETE Book'
+    )
+  })
+})
+
+describe('PurchaseHistories Service Test', () => {
+  let purchaseHistorTestId;
+  const newPurchaseHistory = {
```

Comment:
rename `newPurchaseHistory` as `expectedPurchaseHistory`. It is not a new value, it is already existing in the db